### PR TITLE
Apply suggestions after creator staking code review

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

File: pallets/creator-staking/src/types/mod.rs [hunk 1]
```diff
@@ -68,19 +55,23 @@ pub struct CreatorStakeInfo<Balance: HasCompact + MaxEncodedLen> {
     pub total: Balance,
```
Comment (olehmell) [inline pallets/creator-staking/src/types/mod.rs:55]: In conversation you said that renamed it to "StakesInfo", but you renamed "StakesInfo" to "StakesInfoOf" in the PR. I am confused :)

**Review by olehmell — CHANGES_REQUESTED** [review]

File: pallets/creator-staking/src/types/impls.rs [hunk 2]
```diff
@@ -43,38 +66,53 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
...
         } else {
-            self.stakes.push(EraStake::new(value, current_era));
+            self.stakes.try_push(EraStake::new(value, current_era))
+                .expect("qed; too many stakes in StakesInfo");
         }
 
         Ok(())
     }
 
+    /// Stakes some value in the specified era.
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:100]:
```suggestion
    /// Increase existing stake in the specified era.
```

File: pallets/creator-staking/src/types/impls.rs [hunk 3]
```diff
@@ -43,38 +66,53 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
...
+    ///
+    /// User should ensure that given era is either equal or greater than the
+    /// latest available era in the staking info.
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:103]:
```suggestion
    /// latest available era in the `StakesInfo`.
```

File: pallets/creator-staking/src/types/impls.rs [hunk 4]
```diff
@@ -43,38 +66,53 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
...
+    /// The following example demonstrates how internal vector changes when `stake` is called:
+    ///
+    /// `stakes: [<5, 1000>, <7, 1300>]`
+    /// * `stake(7, 100)` will result in `[<5, 1000>, <7, 1400>]`
+    /// * `stake(9, 200)` will result in `[<5, 1000>, <7, 1400>, <9, 1600>]`
+    pub(crate) fn increase_stake(&mut self, current_era: EraIndex, value: Balance) -> Result<(), &str> {
+        self.change_stake(current_era, value, |x, y| x.saturating_add(y))
+    }
+
     /// Unstakes some value in the specified era.
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:116]:
```suggestion
    /// Decrease existing stake in the specified era.
```

Comment (siman) [reply pallets/creator-staking/src/types/impls.rs:116]: Also update `staking info` to ``StakesInfo``

File: pallets/creator-staking/src/types/impls.rs [hunk 5]
```diff
@@ -94,25 +132,7 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
     ///
     /// Note that if no unclaimed eras remain, vector will be cleared.
     pub(crate) fn unstake(&mut self, current_era: EraIndex, value: Balance) -> Result<(), &str> {
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:134]:
```suggestion
    pub(crate) fn decrease_stake(&mut self, current_era: EraIndex, value: Balance) -> Result<(), &str> {
```

File: pallets/creator-staking/src/types/impls.rs [hunk 6]
```diff
@@ -43,38 +66,53 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
...
     }
 
+    /// Stakes some value in the specified era.
+    ///
+    /// User should ensure that given era is either equal or greater than the
+    /// latest available era in the staking info.
+    ///
+    /// # Example
+    ///
+    /// The following example demonstrates how internal vector changes when `stake` is called:
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:107]:
```suggestion
    /// The following example demonstrates how internal vector changes when `increase_stake()` is called:
```

File: pallets/creator-staking/src/types/impls.rs [hunk 7]
```diff
@@ -43,38 +66,53 @@ impl<Balance: AtLeast32BitUnsigned + Copy + MaxEncodedLen> StakerInfo<Balance> {
...
+    ///
+    /// `stakes: [<5, 1000>, <7, 1300>]`
+    /// * `stake(7, 100)` will result in `[<5, 1000>, <7, 1400>]`
+    /// * `stake(9, 200)` will result in `[<5, 1000>, <7, 1400>, <9, 1600>]`
```
Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:111]:
```suggestion
    /// `increase_stake: [<5, 1000>, <7, 1300>]`
    /// * `increase_stake(7, 100)` will result in `[<5, 1000>, <7, 1400>]`
    /// * `increase_stake(9, 200)` will result in `[<5, 1000>, <7, 1400>, <9, 1600>]`
```

Comment (siman) [inline pallets/creator-staking/src/types/impls.rs:111]:
```suggestion
    /// * `increase_stake(7, 100)` will result in `[<5, 1000>, <7, 1400>]`
    /// * `increase_stake(9, 200)` will result in `[<5, 1000>, <7, 1400>, <9, 1600>]`
```

File: pallets/creator-staking/src/lib.rs [hunk 8]
```diff
@@ -49,27 +50,37 @@ pub mod pallet {
...
 
         /// Maximum number of unique stakers per creator.
         #[pallet::constant]
         type MaxNumberOfStakersPerCreator: Get<u32>;
 
-        /// Max number of unique `EraStake` values that can exist for a `(staker, creator)`
+        /// Max number of unique `EraStake` items that can exist for a `(staker, creator)`
         /// pairing. When stakers claims rewards, they will either keep the number of
         /// `EraStake` values the same or they will reduce them by one. Stakers cannot add
         /// an additional `EraStake` value by calling `bond() & stake()` or `unbond() & unstake()` if they've
```
Comment (siman) [inline pallets/creator-staking/src/lib.rs:86]: Get rid of `bond()` in docs. We do not have it anymore.

File: pallets/creator-staking/src/functions.rs [hunk 9]
```diff
@@ -118,35 +84,36 @@ impl<T: Config> Pallet<T> {
...
-                // One spot should remain for compounding reward claim call
-                staker_info.len() < T::MaxEraStakeValues::get(),
-                Error::<T>::TooManyEraStakeValues
-            );
+            .increase_stake(current_era, desired_amount)
+            .map_err(|_| Error::<T>::CannotChangeStakeInPastEra)?;
+
+        Self::ensure_max_era_stake_items_not_exceeded(staker_info)?;
+
         ensure!(
```
Comment (siman) [inline pallets/creator-staking/src/functions.rs:110]: Add a comment why we do this check after corrections to the stake amount

File: pallets/creator-staking/src/inflation.rs [hunk 10]
```diff
@@ -9,25 +9,31 @@ type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
...
         // Pre-calculate balance which will be deposited for each beneficiary
         let stakers_balance = distro_params.stakers_percent * block_reward.peek();
         let creators_balance = distro_params.creators_percent * block_reward.peek();
+        let treasury_balance = distro_params.treasury_percent * block_reward.peek();
 
         // Prepare imbalances
         let (creators_imbalance, remainder) = block_reward.split(creators_balance);
-        let (stakers_imbalance, treasury_imbalance) = remainder.split(stakers_balance);
+        let (stakers_imbalance, remainder) = remainder.split(stakers_balance);
+        let (fixed_treasury_imbalance, treasury_imbalance) = remainder.split(treasury_balance);
```
Comment (siman) [inline pallets/creator-staking/src/inflation.rs:36]: Please give better names to these two treasury balances

File: pallets/creator-staking/src/lib.rs [hunk 11]
```diff
@@ -49,27 +52,37 @@ pub mod pallet {
...
-        type MinimumStakingAmount: Get<BalanceOf<Self>>;
+        type MinimumStake: Get<BalanceOf<Self>>;
 
         // TODO: make it MinimumRemainingRatio
         //  (e.g. 0.1 = 10%, so that account can lock only 90% of its balance)
         /// Minimum amount that should be left on staker account after staking.
         /// Serves as a safeguard to prevent users from locking their entire free balance.
         #[pallet::constant]
-        type MinimumRemainingAmount: Get<BalanceOf<Self>>;
+        type MinimumRemainingFreeBalance: Get<BalanceOf<Self>>;
```
Comment (siman) [inline pallets/creator-staking/src/lib.rs:79]: Some storages in this pallet are called `Max***` but others `Minimum***`. I mean the short versions of the word vs the long version of the similar word.

**Review by siman — CHANGES_REQUESTED** [review]

**Review by siman — APPROVED** [review]

**Pull request merged**